android/src/main/java/im/shimo/react/prompt/RNPromptFragment.java

```diff
@@ -143,22 +180,68 @@ public Dialog createDialog(Context activityContext, Bundle arguments) {
             }
         }
 
+
+        if (arguments.containsKey(ARG_COLOR)) {
+            String color = arguments.getString(ARG_COLOR);
+            if (color != null) {
+                input.setTextColor(Color.parseColor(color));
+            }
+        }
+
         if (arguments.containsKey(ARG_PLACEHOLDER)) {
             input.setHint(arguments.getString(ARG_PLACEHOLDER));
+            if (arguments.containsKey(ARG_PLACEHOLDER_COLOR)) {
+                String placeholderColor = arguments.getString(ARG_PLACEHOLDER_COLOR);
+                if (placeholderColor != null) {
+                    input.setHintTextColor(Color.parseColor(arguments.getString(ARG_PLACEHOLDER_COLOR)));
+                }
+            }
         }
         alertDialog.setView(input, 50, 15, 50, 0);
 
         mInputText = input;
+
+        alertDialog.setOnShowListener(new DialogInterface.OnShowListener()
+        {
+            @Override
+            public void onShow(final DialogInterface dialog)
+            {
+                input.requestFocus();
+                ((InputMethodManager) alertDialog.getContext().getSystemService(Context.INPUT_METHOD_SERVICE)).showSoftInput(input, 0);
+            }
+        });
+
+
+        input.setOnEditorActionListener(new TextView.OnEditorActionListener() {
+            public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
+                if (actionId == EditorInfo.IME_ACTION_DONE) {
+                    alertDialog.getButton(AlertDialog.BUTTON_POSITIVE).performClick();
+                }
+                return false;
+            }
+        });
+
         return alertDialog;
     }
 
     @Override
     public Dialog onCreateDialog(Bundle savedInstanceState) {
         Dialog dialog = this.createDialog(getActivity(), getArguments());
-        dialog.getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_VISIBLE);
         return dialog;
     }
 
+    @Override
+    public void onStart() {
+        super.onStart();
+
+        if (!mButtonColor.isEmpty()) {
```

Comment:
I just tested this and it crashes as `mButtonColor` is `null` when not specifying this prop or when passed an empty string.

In JS `buttonColor: options.buttonColor || null` will set `buttonColor` to null if not specified or if it's an empty string.

Reply:
Oh shoot i forgot to test for null. Thanks so much on this right away.

Reply:
fixed thanks @jeanregisser!